### PR TITLE
Define BDD test steps for extended Type Concept API

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We define the functions corresponding to the BDD test steps introduced for the extended Type Concept API.

## What are the changes implemented in this PR?

New behaviour test steps and minor (mostly cosmetic) fixes for some existing ones.

### CODE REVIEW

File: test/behaviour/concept/type/attributetype/AttributeTypeSteps.java [hunk 1]
```diff
@@ -124,34 +124,66 @@ public void attribute_type_as_value_type_does_not_have_any_regex(String typeLabe
     }
 
     @Then("attribute\\( ?{type_label} ?) get key owners contain:")
-    public void attribute_type_get_owners_as_key_contains(String typeLabel, List<String> ownerLabels) {
+    public void attribute_type_get_owners_as_key_contain(String typeLabel, List<String> ownerLabels) {
```
Comment (dmitrii-ubskii) [inline test/behaviour/concept/type/attributetype/AttributeTypeSteps.java:127]: Drive-by: fix the wording of the existing step methods to align better with the BDD step.

File: test/behaviour/concept/type/relationtype/RelationTypeSteps.java [hunk 2]
```diff
@@ -75,16 +75,26 @@ public void relation_type_get_role_type_is_null(String relationLabel, String rol
...
+        assertEquals(isNull, isNull(tx().concepts().getRelationType(relationLabel).getRelatesOverridden(roleLabel)));
+    }
+
     @When("relation\\( ?{type_label} ?) get role\\( ?{type_label} ?) set label: {type_label}")
     public void relation_type_get_role_type_set_label(String relationLabel, String roleLabel, String newLabel) {
         tx().concepts().getRelationType(relationLabel).getRelates(roleLabel).setLabel(newLabel);
     }
 
-    @When("relation\\( ?{type_label} ?) get role\\( ?{type_label} ?) get label: {type_label}")
+    @Then("relation\\( ?{type_label} ?) get role\\( ?{type_label} ?) get label: {type_label}")
```
Comment (dmitrii-ubskii) [inline test/behaviour/concept/type/relationtype/RelationTypeSteps.java:88]: Drive-by: `get` steps should be `Then`-assertions, not `When`-prerequisites.

File: test/behaviour/concept/type/attributetype/AttributeTypeSteps.java [hunk 3]
```diff
@@ -124,34 +124,66 @@ public void attribute_type_as_value_type_does_not_have_any_regex(String typeLabe
...
 
     @Then("attribute\\( ?{type_label} ?) get key owners do not contain:")
-    public void attribute_type_get_owners_as_key_do_not_contains(String typeLabel, List<String> ownerLabels) {
+    public void attribute_type_get_owners_as_key_do_not_contain(String typeLabel, List<String> ownerLabels) {
         AttributeType attributeType = tx().concepts().getAttributeType(typeLabel);
         Set<String> actuals = attributeType.getOwners(true).map(ThingType::getLabel).map(Label::toString).toSet();
         for (String ownerLabel : ownerLabels) {
             assertFalse(actuals.contains(ownerLabel));
         }
     }
```
Comment (dmitrii-ubskii) [inline test/behaviour/concept/type/attributetype/AttributeTypeSteps.java:141]: The new steps are for the most part copies of the existing ones with the method swapped for the explicit or overridden variation.

File: test/behaviour/concept/type/relationtype/RelationTypeSteps.java [hunk 4]
```diff
@@ -109,6 +119,25 @@ public void relation_type_get_related_roles_do_not_contain(String relationLabel,
         }
     }
 
+    private Set<Parameters.ScopedLabel> relation_type_get_related_explicit_roles_actuals(String relationLabel) {
```
Comment (flyingsilverfin) [inline test/behaviour/concept/type/relationtype/RelationTypeSteps.java:122]: same comment as in `client-java` :)

**Review by flyingsilverfin — APPROVED** [review]
